### ISSUE TITLE
Error compiling project

### ISSUE THREAD

**Issue by WeirdCircumstances** [mikebrady/shairport-sync#925]
Hello everybody!

I'm using shairport-sync for a long time now and I want to say thank you for all this awesome work here!

Today I got some errors compiling your project successfully. I looked for a while, but was not able to find any solution. I think the fault is on my side, but at this point I can't go further. This is what I got:

```
pi@rune:~/shairport-sync/temp/shairport-sync $ make -j4
make  all-recursive
make[1]: Entering directory '/home/pi/shairport-sync/temp/shairport-sync'
Making all in man
make[2]: Entering directory '/home/pi/shairport-sync/temp/shairport-sync/man'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/pi/shairport-sync/temp/shairport-sync/man'
make[2]: Entering directory '/home/pi/shairport-sync/temp/shairport-sync'
depbase=`echo alac.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\"  -g -O2 -MT alac.o -MD -MP -MF $depbase.Tpo -c -o alac.o alac.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo FFTConvolver/AudioFFT.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
g++ -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\" -std=c++11 -g -O2 -MT FFTConvolver/AudioFFT.o -MD -MP -MF $depbase.Tpo -c -o FFTConvolver/AudioFFT.o FFTConvolver/AudioFFT.cpp &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo FFTConvolver/FFTConvolver.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
g++ -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\" -std=c++11 -g -O2 -MT FFTConvolver/FFTConvolver.o -MD -MP -MF $depbase.Tpo -c -o FFTConvolver/FFTConvolver.o FFTConvolver/FFTConvolver.cpp &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo FFTConvolver/Utilities.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
g++ -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\" -std=c++11 -g -O2 -MT FFTConvolver/Utilities.o -MD -MP -MF $depbase.Tpo -c -o FFTConvolver/Utilities.o FFTConvolver/Utilities.cpp &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo FFTConvolver/convolver.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
g++ -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\" -std=c++11 -g -O2 -MT FFTConvolver/convolver.o -MD -MP -MF $depbase.Tpo -c -o FFTConvolver/convolver.o FFTConvolver/convolver.cpp &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo shairport.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\"  -g -O2 -MT shairport.o -MD -MP -MF $depbase.Tpo -c -o shairport.o shairport.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo rtsp.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\"  -g -O2 -MT rtsp.o -MD -MP -MF $depbase.Tpo -c -o rtsp.o rtsp.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo mdns.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\"  -g -O2 -MT mdns.o -MD -MP -MF $depbase.Tpo -c -o mdns.o mdns.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo common.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\"  -g -O2 -MT common.o -MD -MP -MF $depbase.Tpo -c -o common.o common.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo rtp.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\"  -g -O2 -MT rtp.o -MD -MP -MF $depbase.Tpo -c -o rtp.o rtp.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo player.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\"  -g -O2 -MT player.o -MD -MP -MF $depbase.Tpo -c -o player.o player.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo audio.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\"  -g -O2 -MT audio.o -MD -MP -MF $depbase.Tpo -c -o audio.o audio.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo loudness.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\"  -g -O2 -MT loudness.o -MD -MP -MF $depbase.Tpo -c -o loudness.o loudness.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo activity_monitor.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\"  -g -O2 -MT activity_monitor.o -MD -MP -MF $depbase.Tpo -c -o activity_monitor.o activity_monitor.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo apple_alac.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
g++ -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\" -std=c++11 -g -O2 -MT apple_alac.o -MD -MP -MF $depbase.Tpo -c -o apple_alac.o apple_alac.cpp &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo mdns_avahi.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\"  -g -O2 -MT mdns_avahi.o -MD -MP -MF $depbase.Tpo -c -o mdns_avahi.o mdns_avahi.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo audio_alsa.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.    -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\"  -g -O2 -MT audio_alsa.o -MD -MP -MF $depbase.Tpo -c -o audio_alsa.o audio_alsa.c &&\
mv -f $depbase.Tpo $depbase.Po
g++ -Wno-multichar -Wall -Wextra -Wno-clobbered -Wno-psabi -pthread -DSYSCONFDIR=\"/etc\" -std=c++11 -g -O2   -o shairport-sync shairport.o rtsp.o mdns.o common.o rtp.o player.o alac.o audio.o loudness.o activity_monitor.o apple_alac.o mdns_avahi.o   audio_alsa.o         FFTConvolver/AudioFFT.o FFTConvolver/FFTConvolver.o FFTConvolver/Utilities.o FFTConvolver/convolver.o         -lsndfile -lasound -lavahi-common -lavahi-client -lsoxr -lssl -lcrypto -L/usr/local/lib -lalac -lconfig -lpopt -lm -lpthread -lrt 
/usr/bin/ld: FFTConvolver/convolver.o: in function `convolver_init':
/home/pi/shairport-sync/temp/shairport-sync/FFTConvolver/convolver.cpp:23: undefined reference to `die'
/usr/bin/ld: /home/pi/shairport-sync/temp/shairport-sync/FFTConvolver/convolver.cpp:26: undefined reference to `die'
/usr/bin/ld: /home/pi/shairport-sync/temp/shairport-sync/FFTConvolver/convolver.cpp:53: undefined reference to `debug'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:737: shairport-sync] Error 1
make[2]: Leaving directory '/home/pi/shairport-sync/temp/shairport-sync'
make[1]: *** [Makefile:870: all-recursive] Error 1
make[1]: Leaving directory '/home/pi/shairport-sync/temp/shairport-sync'
make: *** [Makefile:594: all] Error 2
```

I compiled it on my raspberry as well as on my debian server, with same result. Everything is fresh cloned from here. Last month it was still working.

Thanks for any feedback!

**Comment by mikebrady** [comment]
Thanks for the report. It's a compilation bug, sorry. Will fix today.

**Comment by mikebrady** [comment]
Fixed, I think, in Version 3.3.4!
